linfa-kernel/src/lib.rs

```diff
@@ -40,18 +30,20 @@ pub enum KernelInner<F: Float> {
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-pub struct Kernel<R: Records>
+pub struct KernelBase<R: Records, K1: Inner, K2: Inner>
 where
     R::Elem: Float,
+    K1::Elem: Float,
+    K2::Elem: Float,
 {
     #[cfg_attr(
         feature = "serde",
         serde(bound(
-            serialize = "KernelInner<R::Elem>: Serialize",
-            deserialize = "KernelInner<R::Elem>: Deserialize<'de>"
+            serialize = "KernelInner<K1, K2>: Serialize",
+            deserialize = "KernelInner<K1, K2>: Deserialize<'de>"
```

Comment:
I'm not familiar with `serde` at all so I don't know if this is the right way to do this

Reply:
this is correct, `serde` needs the signatures to understand the type